src/frontend/sharedComponents/BottomSheetModal/Content.tsx

```diff
@@ -0,0 +1,160 @@
+import * as React from 'react';
+import {StyleSheet, TextStyle, ScrollView, View} from 'react-native';
+import {TouchableHighlight} from '@gorhom/bottom-sheet';
+import {useDimensions} from '@react-native-community/hooks';
+import {UIActivityIndicator} from 'react-native-indicators';
+
+import {LIGHT_BLUE, MAGENTA, COMAPEO_BLUE, RED, WHITE} from '../../lib/styles';
+import {Button} from '../Button';
+import {Text} from '../Text';
+import {useBottomSheetModalProperties} from './BottomSheetModalPropertiesContext';
+
+const MINIMUM_SHEET_HEIGHT = 400;
+const LOADING_INDICATOR_HEIGHT = 40;
+
+interface BaseActionButtonConfig {
+  onPress: () => void;
+  text: React.ReactNode;
+  variation: 'filled' | 'outlined';
+  icon?: React.ReactNode;
+}
+
+interface PrimaryActionButtonConfig extends BaseActionButtonConfig {
+  dangerous?: boolean;
+  variation: 'filled';
+}
+
+interface SecondaryActionButtonConfig extends BaseActionButtonConfig {
+  variation: 'outlined';
+}
+
+export type ActionButtonConfig =
+  | PrimaryActionButtonConfig
+  | SecondaryActionButtonConfig;
+
+export interface Props extends React.PropsWithChildren {
+  buttonConfigs: ActionButtonConfig[];
+  description?: React.ReactNode;
+  descriptionStyle?: TextStyle;
+  icon?: React.ReactNode;
+  loading?: boolean;
+  title: React.ReactNode;
+  titleStyle?: TextStyle;
+}
+
+export const Content = ({
```

Comment:
part of me thinks that maybe we should have 2 separate components exported: a fullscreen and a dynamic variant. can't imagine cases where a consuming component would switch between the two.

can be done as a follow-up though